### PR TITLE
[Refactor] demo for object detection

### PR DESCRIPTION
**Pull request by chongruo**
coding refactoring for object detection

### CODE REVIEW

File: autogluon/task/object_detection/dataset/__init__.py [hunk 1]
```diff
@@ -6,16 +6,20 @@
...
+
+        splits = [('', index_file_name)]
+        return CustomVOCDetection(root, splits, name, classes, **kwargs)
+
+    elif format=='coco':
+        if dataset_name=="coco":
+            logger.info(">>> create dataset(COOC format)")
+            return COCO(*args, **kwargs)
+        else:
+            raise NotImplementedError('Dataset: {} not implemented.'.format(dataset))
```
Comment (Innixma) [inline autogluon/task/object_detection/dataset/__init__.py]: dataset is not a variable

File: autogluon/task/object_detection/dataset/__init__.py [hunk 2]
```diff
@@ -6,16 +6,20 @@
...
     else:
         raise NotImplementedError('Dataset: {} not implemented.'.format(dataset))
```
Comment (Innixma) [inline autogluon/task/object_detection/dataset/__init__.py]: dataset is not a variable

File: autogluon/task/object_detection/object_detection.py [hunk 3]
```diff
@@ -119,6 +120,8 @@ def fit(dataset='voc',
             pass
 
         nthreads_per_trial = get_cpu_count() if nthreads_per_trial > get_cpu_count() else nthreads_per_trial
+        if ngpus_per_trial > get_gpu_count():
+            logger.warning("The number of requested GPUs is greater than the number availabe GPUs.")
```
Comment (Innixma) [inline autogluon/task/object_detection/object_detection.py]: typo

File: autogluon/task/object_detection/dataset/__init__.py (hunk 1 above)
Comment (chongruo) [reply autogluon/task/object_detection/dataset/__init__.py]: Thanks. fixed

**Comment by szha** [comment]
Job PR-154-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-154/2/index.html

**Pull request merged**